### ISSUE TITLE
Implement Loss Functions

### ISSUE THREAD

**Issue by ayushm-agrawal** [ayushm-agrawal/PySaber#5]
**What is the Issue**
Currently, we use PyTorch's built-in loss functions to test our networks. We would like to implement our own versions of the following loss classes.

- [ ] Log Softmax
- [ ] NLLLoss
- [ ] Cross-Entropy Loss

**Where to fix**
All the loss classes will go under **saber/nn/modules/loss.py**
